File: PetMedicalCare/BackEnd/SoftPetBusiness/src/main/java/pe/edu/pucp/softpet/bo/RolUsuarioBo.java
package pe.edu.pucp.softpet.bo;

import java.util.ArrayList;
import pe.edu.pucp.softpet.dao.RolUsuarioDao;
import pe.edu.pucp.softpet.daoImp.RolUsuarioDaoImpl;
import pe.edu.pucp.softpet.dto.usuarios.RolDto;
import pe.edu.pucp.softpet.dto.usuarios.RolUsuarioDto;
import pe.edu.pucp.softpet.dto.usuarios.UsuarioDto;

public class RolUsuarioBo {

    private final RolUsuarioDao rolUsuarioDao;

    public RolUsuarioBo() {
        this.rolUsuarioDao = new RolUsuarioDaoImpl();
    }

    public Integer insertar(int rolId, int usuarioId, boolean activo) {

        RolUsuarioDto rolUsuario = new RolUsuarioDto();

        RolDto rol = new RolDto();
        UsuarioDto usuario = new UsuarioDto();

        rol.setRolId(rolId);
        usuario.setUsuarioId(usuarioId);

        rolUsuario.setRol(rol);
        rolUsuario.setUsuario(usuario);
        rolUsuario.setActivo(activo);

        return this.rolUsuarioDao.insertar(rolUsuario);
    }

    public Integer modificar(int rolUsuarioId, int rolId, int usuarioId, boolean activo) {

        RolUsuarioDto rolUsuario = new RolUsuarioDto();

        RolDto rol = new RolDto();
        UsuarioDto usuario = new UsuarioDto();

        rol.setRolId(rolId);
        usuario.setUsuarioId(usuarioId);

        rolUsuario.setRolUsuarioId(rolUsuarioId);
        rolUsuario.setRol(rol);
        rolUsuario.setUsuario(usuario);
        rolUsuario.setActivo(activo);

        return this.rolUsuarioDao.modificar(rolUsuario);
    }

    public Integer eliminar(Integer rolUsuarioId) {
        RolUsuarioDto rolUsuarioDto = new RolUsuarioDto();
        rolUsuarioDto.setRolUsuarioId(rolUsuarioId);
        return this.rolUsuarioDao.eliminar(rolUsuarioDto);
    }

    public RolUsuarioDto obtenerPorID(int rolUsuarioId) {
        RolUsuarioDto rolUsuarioDto = new RolUsuarioDto();
        rolUsuarioDto.setRolUsuarioId(rolUsuarioId);
        return this.rolUsuarioDao.obtenerPorId(rolUsuarioId);
    }

    public ArrayList<RolUsuarioDto> listarTodos() {
        return this.rolUsuarioDao.listarTodos();
    }
}

File: PetMedicalCare/FrontEnd/SoftPetWA/SoftPetBussiness/SOAPBo/Seguridad/RolUsuarioBo.cs
using SoftPetBussiness.RolUsuarioClient;
using System.Collections.Generic;
using System.Linq;

namespace SoftPetBusiness
{
    public class RolUsuarioBO
    {
        private RolesUsuarioClient clienteSOAP;

        public RolUsuarioBO()
        {
            this.clienteSOAP = new RolesUsuarioClient();
        }

        // Insertar relación Rol-Usuario
        public int Insertar(int rolId, int usuarioId, bool activo)
        {
            return this.clienteSOAP.insertar(rolId, usuarioId, activo);
        }

        // Modificar relación Rol-Usuario
        public int Modificar(int rolUsuarioId, int rolId, int usuarioId, bool activo)
        {
            return this.clienteSOAP.modificar(rolUsuarioId, rolId, usuarioId, activo);
        }

        // Eliminar relación Rol-Usuario
        public int Eliminar(int rolUsuarioId)
        {
            return this.clienteSOAP.eliminar(rolUsuarioId);
        }

        // Obtener RolUsuario por ID
        public rolUsuarioDto ObtenerPorId(int rolUsuarioId)
        {
            return this.clienteSOAP.obtenerPorId(rolUsuarioId);
        }

        // Listar todas las relaciones Rol-Usuario
        public List<rolUsuarioDto> ListarTodos()
        {
            return this.clienteSOAP.listarTodos().ToList<rolUsuarioDto>();
        }

        // Insertar usando DTO
        public int Insertar(rolUsuarioDto rolUsuario)
        {
            return this.clienteSOAP.insertar(
                rolUsuario.rol.rolId,
                rolUsuario.usuario.usuarioId,
                rolUsuario.activo
            );
        }

        // Modificar usando DTO
        public int Modificar(rolUsuarioDto rolUsuario)
        {
            return this.clienteSOAP.modificar(
                rolUsuario.rolUsuarioId,
                rolUsuario.rol.rolId,
                rolUsuario.usuario.usuarioId,
                rolUsuario.activo
            );
        }
    }
}
